### PR TITLE
feat: add update sidebar for novu providers

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What change does this PR introduce?

Add info for Novu providers in update provider sidebar

### Other information (Screenshots)
Screenshot 2023-07-05 at 13 43 17

### CODE REVIEW

File: apps/web/src/pages/integrations/UpdateProviderPage.tsx [hunk 1]
```diff
@@ -29,10 +37,22 @@ interface IProviderForm {
   identifier: string;
 }
 
+const Info = ({ provider, environments }) => (
```
Comment (davidsoderberg) [inline apps/web/src/pages/integrations/UpdateProviderPage.tsx:40]: Maybe this should be in its own file even if it is only used here and it feels like this is the only place it will be used it?

File: apps/web/src/pages/integrations/UpdateProviderPage.tsx [hunk 2]
```diff
@@ -152,6 +172,45 @@ export function UpdateProviderPage() {
     return null;
   }
 
+  if (
+    SmsProviderIdEnum.Novu === selectedProvider.providerId ||
+    EmailProviderIdEnum.Novu === selectedProvider.providerId
+  ) {
+    return (
```
Comment (davidsoderberg) [inline apps/web/src/pages/integrations/UpdateProviderPage.tsx:179]: Tried to reuse as many components as possible

File: packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.usecase.ts [hunk 3]
```diff
@@ -134,7 +137,19 @@ export class GetNovuIntegration {
       from: process.env.NOVU_SMS_INTEGRATION_SENDER,
     };
 
-    return item;
+    return this.mapNovuItem(item, command);
+  }
+
+  private mapNovuItem(
```
Comment (davidsoderberg) [inline packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.usecase.ts:143]: Adding these as values until the exists in the database.

**Comment by davidsoderberg** [comment]
The images for novu sms provider did not exists before so created them now :)

File: packages/application-generic/src/usecases/get-decrypted-integrations/get-decrypted-integrations.usecase.ts [hunk 4]
```diff
@@ -50,19 +56,58 @@ export class GetDecryptedIntegrations {
...
+          environmentId: command.environmentId,
+          userId: command.userId,
+        })
+      );
+
+      return novuIntegration ? [novuIntegration] : [];
     }
 
-    const novuIntegration = await this.getNovuIntegration.execute(
+    const novuSmsIntegration = await this.getNovuIntegration.execute(
```
Comment (davidsoderberg) [inline packages/application-generic/src/usecases/get-decrypted-integrations/get-decrypted-integrations.usecase.ts:85]: This replaces the hardcoded providers in frontend

File: packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts [hunk 5]
```diff
@@ -7,4 +7,8 @@ export class GetNovuIntegrationCommand extends EnvironmentWithUserCommand {
   @IsOptional()
   @IsEnum(ChannelTypeEnum)
   channelType?: ChannelTypeEnum;
+
+  @IsOptional()
+  @IsBoolean()
+  ignoreActiveCount? = false;
```
Comment (p-fernandez) [inline packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts:13]:
```suggestion
  ignoreActiveCount = false;
```
Not sure if this can be done here but if setting the default value you don't need to add the question mark.

File: apps/web/src/pages/integrations/useProviders.ts [hunk 6]
```diff
@@ -72,17 +76,17 @@ function initializeProvidersByIntegration(integrations: IntegrationEntity[]): II
     fcmFallback(integrationItem, clonedCredentials);
 
     return {
-      providerId: providerItem.id,
+      providerId: integrationItem.providerId || providerItem?.id,
```
Comment (p-fernandez) [inline apps/web/src/pages/integrations/useProviders.ts:79]: I do not understand why in this return some of the fallbacks are prioritised by `integrationItem` and others `providerItem`. Could you share the reasoning of it? I am confused, sorry. 😞

File: apps/web/src/pages/integrations/UpdateProviderPage.tsx (hunk 1 above)
Comment (p-fernandez) [reply apps/web/src/pages/integrations/UpdateProviderPage.tsx:42]: There is another PR where I think @LetItRock implements something similar. 
https://github.com/novuhq/novu/blob/e97f83e0482882d1b218c9d6cce71e66ff5571b1/apps/web/src/pages/integrations/components/UpdateIntegrationSidebarHeader.tsx#L41

https://github.com/novuhq/novu/pull/3740

So I agree with you and I push for it to be moved it to its own component so it can be reused for you both. 👏🏻

File: apps/api/src/app/integrations/e2e/create-integration.e2e.ts [hunk 7]
```diff
@@ -23,13 +23,13 @@ describe('Create Integration - /integration (POST)', function () {
...
-    const emailIntegrations = integrations.filter(
+    const emailIntegrations: any[] = integrations.filter(
       (searchIntegration) => searchIntegration.channel === ChannelTypeEnum.EMAIL
     );
 
-    expect(emailIntegrations.length).to.eql(2);
+    expect(emailIntegrations.length).to.eql(3);
 
-    for (const emailIntegration of emailIntegrations) {
+    for (const emailIntegration of emailIntegrations.splice(1, 1)) {
```
Comment (p-fernandez) [inline apps/api/src/app/integrations/e2e/create-integration.e2e.ts:32]: I am sorry, can't understand what you try to do here. 😞

Comment (davidsoderberg) [reply apps/api/src/app/integrations/e2e/create-integration.e2e.ts:32]: We do it because the last one will be Novu free provider and we do not like to check it here.

File: apps/web/src/pages/integrations/useProviders.ts (hunk 6 above)
Comment (davidsoderberg) [reply apps/web/src/pages/integrations/useProviders.ts:79]: Because Novus free provider is not a part of the novu providers list and then we need to use it for the integration item we get from the backend.

File: packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts (hunk 5 above)
Comment (davidsoderberg) [reply packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts:13]: It needs to be there otherwise typescript thinks it is always needed to be provided.

File: apps/api/src/app/integrations/e2e/create-integration.e2e.ts (hunk 7 above)
Comment (davidsoderberg) [reply apps/api/src/app/integrations/e2e/create-integration.e2e.ts]: I changed it to make more sense :)

File: apps/api/src/app/integrations/usecases/get-integrations/get-integrations.usecase.ts [hunk 8]
```diff
@@ -13,6 +13,7 @@ export class GetIntegrations {
       GetDecryptedIntegrationsCommand.create({
         organizationId: command.organizationId,
         userId: command.userId,
+        environmentId: command.environmentId,
```
Comment (ainouzgali) [inline apps/api/src/app/integrations/usecases/get-integrations/get-integrations.usecase.ts:16]: Why did you add the environment? As I understand it we do want all integrations, from both environments

Comment (davidsoderberg) [reply apps/api/src/app/integrations/usecases/get-integrations/get-integrations.usecase.ts:16]: because the feature flag usecase requires it :)

Comment (ainouzgali) [reply apps/api/src/app/integrations/usecases/get-integrations/get-integrations.usecase.ts:16]: Oops. Sorry ;)

Comment (p-fernandez) [reply apps/api/src/app/integrations/usecases/get-integrations/get-integrations.usecase.ts:16]: @davidsoderberg mind to leave a TODO comment to remind us to remove once we sunset the pre feature functionality?

File: packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts (hunk 5 above)
Comment (p-fernandez) [reply packages/application-generic/src/usecases/get-novu-integration/get-novu-integration.command.ts:13]: I am so disappointed it behaves differently as the arguments in a function. 😞

**Review by p-fernandez — APPROVED** [review]
